junifer/api/queue_context/htcondor_adapter.py

```diff
@@ -178,7 +188,7 @@ def _check_collect(self, collect: str) -> str:
     def pre_run(self) -> str:
         """Return pre-run commands."""
         fixed = (
-            "#!/bin/bash\n\n"
+            f"#!/bin/{self._shell}\n\n"
```

Comment:
would it be better to do `/bin/env {shell}`?

Reply:
The standard advice that I see in the wild is to use `#!/bin/<shell>`. Is there a particular reason for your suggestion?

Reply:
Running a command through /usr/bin/env has the benefit of looking for whatever the default version of the program is in your current environment.

This way, you don't have to look for it in a specific place on the system, as those paths may be in different locations on different systems. As long as it's in your path, it will find it.

But at the end of the day it isnt really going to make a difference.

Reply:
That's a fair argument if you want to have control over the shell used. Would you say we extend it to `run_{conda,venv}.{bash,zsh}` as well?

Reply:
Yes, i think so, but didnt wanna duplicate the comment.